ringvax/app.py

```diff
@@ -99,21 +143,68 @@ def app():
+    sims = []
+    with st.spinner("Running simulation..."):
+        tic = time.perf_counter()
+        for i in range(nsim):
+            sims.append(Simulation(params=params, seed=seed + i))
+            sims[-1].run()
+        toc = time.perf_counter()
+
+    st.write(
+        f"Ran {nsim} simulations in {format_duration(toc - tic)} with an $R_0$ of {infectious_duration * infection_rate:.2f} (the product of the average duration of infection and the infectious rate)."
     )
 
-    s = Simulation(params=params, seed=seed)
-    s.run()
+    tab1, tab2 = st.tabs(["Simulation summary", "Per-simulation results"])
+    with tab1:
+        sim_df = get_all_person_properties(sims)
+
+        pr_control = prob_control_by_gen(sim_df, control_generations)
+        st.header(
+            f"Probability of control: {pr_control:.2f}",
+            help=f"The probability that there are no infections in the {format_control_gens(control_generations)}, or equivalently that the {format_control_gens(control_generations - 1)} do not produce any further infections.",
+        )
 
-    st.header("Graph of infections")
-    st.graphviz_chart(make_graph(s))
+        st.header("Number of infections")
+        st.write(
+            "Distribution of the number of infections across the simulated generations."
+        )
+        st.altair_chart(
+            alt.Chart(get_outbreak_size_df(sim_df))
+            .mark_bar()
+            .encode(
+                x=alt.X("size:Q", bin=True, title="Number of infections"),
+                y=alt.Y("count()", title="Count"),
+            )
+        )
+
+        st.header("Summary of dynamics")
+        infection = summarize_infections(sim_df)
+        st.write(
+            f"In these simulations, the average duration of infectiousness was {infection['mean_infectious_duration'][0]:.2f} and $R_e$ was {infection['mean_n_infections'][0]:.2f}"
+        )
 
-    st.header("Raw results")
-    for id, content in s.infections.items():
-        st.text(id)
-        st.text(content)
+        st.write(
+            "The following table provides summaries of marginal probabilities regarding detection. Aside from the marginal probability of active detection, these are the observed probabilities that any individual is detected in this manner. The marginal probability of active detection excludes index cases, which are not eligible for active detection."
+        )
+        detection = summarize_detections(sim_df)
+        st.dataframe(
+            detection.select(pl.col(col).round(2) for col in detection.columns).rename(
+                {
+                    "prob_detect": "Any detection",
+                    "prob_active": "Active detection",
+                    "prob_passive": "Passive detection",
+                    "prob_detect_before_infectious": "Detection before onset of infectiousness",
+                }
+            )
+        )
+
+    with tab2:
```

Comment:
Unfortunately, any time you change which simulation you're showing, it re-runs all the simulations

I was reading about streamlit reactivity; I think I can fix this, but let's do it in another PR

Reply:
#35